detectron2/utils/patch_instance.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import importlib.util
+import os
+import sys
+import tempfile
+import torch
+
+from detectron2.structures import Boxes, Instances  # noqa F401
+
+
+def export_torchscript_with_patch_instance(model, fields):
+    with tempfile.NamedTemporaryFile(mode="w", encoding="utf-8", suffix=".py", delete=False) as f:
+        cls_name, s = gen_class(fields)
+        f.write(s)
+        f.flush()
+        f.close()
+
+        module = _import(f.name)
+        new_instance = getattr(module, cls_name)
+        _ = torch.jit.script(new_instance)
+
+        Instances.__torch_script_class__ = True
+        Instances._jit_override_qualname = torch._jit_internal._qualified_name(new_instance)
+        return torch.jit.script(model), new_instance
+
+
+def _typing_imports_str():
+    lines = []
+    lines.append("import torch")
+    lines.append("from torch import Tensor")
+    lines.append("import typing")
+    lines.append("from typing import *")
+    lines.append("")
+    lines.append("")
+    lines.append("from detectron2.structures import Boxes")
+    lines.append("")
+    lines.append("")
+    return os.linesep.join(lines)
+
+
+_counter = 0  # TODO replace by a hash?
+
+
+def _gen_class(fields):
+    def indent(level, s):
+        return " " * 4 * level + s
+
+    lines = []
+
+    cls_name = "Instance{}".format(_counter)
+    lines.append("class {}:".format(cls_name))
+    lines.append("")
+
+    lines.append(indent(1, "def __init__(self, image_size: Tuple[int, int]):"))
+    lines.append(indent(2, "self.image_size = image_size"))
```

Comment:
similarly it can use multi-line string and f-string to improve readability

Reply:
done